src/PartialModelSelection.cpp

```diff
@@ -1,202 +1,199 @@
-Exceptions: none?
-Note: none
-*/
-double ModelSelectionMap::getNewPenaltyList()
-{
-
-  //If the map of tested pairs is empty, query penaltyQuery 0 first.  
-  if(!hasModelsInserted())
-     {
-       return EMPTY_MAP_QUERY; //?
-     }
-
-  //penaltyModelpenaltyModelPair smallestPair = penaltyModelMap.begin()->first;
-  
-  //for (std::map<double,Model>::iterator it=penaltyModelMap.begin(); it!=penaltyModelMap.end(); ++it)
-    //  std::cout << it->first << " => " << it->second.model_size << '\n';   
-   return 0;
-     
-    
-  
-}
-
-//Method to display the currently stored pairs in the map. 
-void ModelSelectionMap::displayMap() {
-  std::cout << "#################################\n" << "Current Map Display\n" << "#################################\n"; 
-  for (std::map<double,Model>::iterator it=penaltyModelMap.begin(); it!=penaltyModelMap.end(); ++it)
-      std::cout << it->first << " => " << it->second.model_size << '\n';
-
-}
-
-
- bool ModelSelectionMap::hasModelsInserted(){
-    //Check to see if only the default entry is in the map.
-    return insertedModels > 0;
-
- }
-
-
-//MinimizeResult Method Implementations
-
-//MinimizeResult default constructor to be used when no models have been inserted. 
-MinimizeResult::MinimizeResult(){
- //Initialize a default result, which has 1 selected, but unsure.
- optimalModels.first = -1;
- optimalModels.second = INFINITY;
- certain = false;
-
-}
-
-MinimizeResult::MinimizeResult(std::pair<double, double> inputModels)
-   {
-       //Check for a valid range of models. 
-       if(!isValidRange(inputModels))
-          {
-           throw std::out_of_range("Invalid range inputted to MinimizeResult creation."); 
-          }
-
-       else
-          {
-           //Check if our penaltyQueryRange is a solved point (Where the beginning and end are identical.)
-           if(optimalModels.first == optimalModels.second)
-              {
-               certain = true;
-              }
-            else{
-               certain = false;  
-
-            }
-            optimalModels.first = inputModels.first;
-            optimalModels.second = inputModels.second;
-
-          }
-       
-      
-   }
-
-
-bool MinimizeResult::isValidRange(std::pair<double,double> optimalModels)
-   {
-    //Tie a model size and pairing to a penaltyQuery range.
-    if(optimalModels.first < 0 || optimalModels.first < optimalModels.second)
-       {
-        return false;   
-       }
-    return true;
-   }
-
-
-
-std::pair<int, int> ModelSelectionMap::solver(double penaltyQuery)
-{
-   auto tempPair = std::make_pair<int, bool>(4, true);
-   std::cout << "Testing solver!\n";
-   //Temporary stub return
-   return tempPair;
-}
-
-
-
-
+//System Includes and Libraries
+#include <iostream>
+#include <map>
+#include <stdexcept>
+#include <vector>
+#include <math.h>
+#include <string>
+//Local includes
+#include "PartialModelSelection.hpp"
+#include "ModelSelectionUtils.hpp"
```

Comment:
why are there two different hpp files? can you please combine them into a single header? that is more typical (less confusing)

Reply:
Sure, but I can still segregate method implementations into two separate .cpp files right? That helps from keeping one file too long.